saltboot/salt.go

```diff
@@ -193,7 +194,26 @@ func SaltMinionRunRequestHandler(w http.ResponseWriter, req *http.Request) {
 	}
 
 	grainConfigPath := baseDir + "/etc/salt/grains"
+	prewarmedRolesPath := baseDir + "/etc/salt/prewarmed_roles"
 	if isGrainsConfigNeeded(grainConfigPath) {
+		// Check if prewarmed roles exist, and add them to the roles before generating the file
+		if shouldAppendPrewarmedRoles(prewarmedRolesPath) {
+			file, err := os.Open(prewarmedRolesPath)
```

Comment:
We could use ioutil.ReadFile

Reply:
I suppose we could. This seems simpler for reading one line of text at a time, and has been tested. Let me know if you want me to try changing it.